console/frontend/src/main/frontend/src/app/components/th-sortable.directive.ts

```diff
@@ -0,0 +1,76 @@
+import { Directive, ElementRef, EventEmitter, Input, OnInit, Output, QueryList } from '@angular/core';
+
+export type SortDirection = 'asc' | 'desc' | null;
+export type SortEvent = {
+  column: string;
+  direction: SortDirection;
+}
+
+export function updateSortableHeaders(headers: QueryList<ThSortableDirective>, column: string) {
+  for (const header of headers) {
+    if (header.sortable !== column) {
+      header.updateDirection(null);
+    }
+  }
+}
+
+const compare = (v1: string | number, v2: string | number) => (v1 < v2 ? -1 : v1 > v2 ? 1 : 0);
+export function basicTableSort<T extends Record<string, any>>(arr: T[], headers: QueryList<ThSortableDirective>, { column, direction }: SortEvent): T[] {
+  updateSortableHeaders(headers, column);
+
+  if (direction == null || column == '')
+    return arr;
+
+  return [...arr].sort((a, b) => {
+    const order = compare(a[column], b[column]);
+    return direction === 'asc' ? order : -order;
+  });
+}
+
+@Directive({
+  selector: 'th[sortable]',
+  host: {
+    '(click)': 'nextSort()',
+  }
+})
+export class ThSortableDirective implements OnInit {
+  @Input() sortable: string = '';
+  @Input() direction: SortDirection = null;
+  @Output() onSort = new EventEmitter<SortEvent>();
+
+  private nextSortOption: Record<string, SortDirection> = {
+    '': 'asc',
+    asc: 'desc',
+    desc: null
+  }
```

Comment:
Would this be possible? If so, I think it is more logical to have them be the same type. 

```suggestion
  private nextSortOption: Record<SortDirection, SortDirection> = {
    null: 'asc',
    asc: 'desc',
    desc: null
  }
```

Reply:
Sadly `null` cant be an object key, thats the only reason I used an empty string and a null check instead